hotshot-state-prover/src/service.rs

```diff
@@ -315,7 +368,37 @@ pub async fn submit_state_and_proof(
     // prepare the input the contract call and the tx itself
     let proof: ParsedPlonkProof = proof.into();
     let new_state: ParsedLightClientState = public_input.into();
-    let tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    let mut tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    // TODO: (alex) centralize this to a dedicated ethtxmanager with gas control
+    // frugal gas price: set to SafeGasPrice based on live price oracle
+    // safe to be included with low priority position in a block.
+    // ignore this if etherscan fail to return
+    if let Ok(res) =
+        reqwest::get("https://api.etherscan.io/api?module=gastracker&action=gasoracle").await
```

Comment:
This needs to be a configurable URL

Reply:
Ideally it should be an optional URL. The logic really isn't that complicated if the L1 doesn't have a gas price oracle: we just skip this block if the configured URL is none (using the default gas price estimation from ethers.rs). This would be a much better behavior for Sepolia than following completely unrelated gas trends from mainnet

Reply:
Sounds good. Just to clarify, we don't care _that_ much about testnet gas, but it is still a scarce resource, and moreover the mainnet and testnet gas prices are completely unrelated, so we might have transactions failing/not getting mined if testnet gas price is spiking and we are setting too low of a price, or if mainnet gas price is spiking we might burn through our sep eth for no reason

Reply:
refactored in [384fc7f](https://github.com/EspressoSystems/espresso-sequencer/pull/2148/commits/384fc7f9c80eb0221d0057e4efee95d83b5fb831)

1. only invoke gas oracle if the provider is detected to be on mainnet
2. use blocknative code (directly copied from ethers-rs, with minimal changes to make it in sync with the latest feed data format, see https://github.com/EspressoSystems/espresso-sequencer/pull/2148#issuecomment-2415359785). The URL is still hardcoded, but we can easily switch to a different price oracle in the future if we want.